python/dolfinx/fem/assemble.py

```diff
@@ -11,36 +11,39 @@
 import functools
 import typing
 import warnings
+from collections.abc import Iterable
 
 import numpy as np
+import numpy.typing as npt
 
 import dolfinx
 from dolfinx import cpp as _cpp
 from dolfinx import la
 from dolfinx.cpp.fem import pack_coefficients as _pack_coefficients
 from dolfinx.cpp.fem import pack_constants as _pack_constants
+from dolfinx.fem import IntegralType
 from dolfinx.fem.bcs import DirichletBC
 from dolfinx.fem.forms import Form
 
 
 def pack_constants(
     form: typing.Union[Form, typing.Sequence[Form]],
 ) -> typing.Union[np.ndarray, typing.Sequence[np.ndarray]]:
```

Comment:
Should we make the type-hint more specific, and use `typing.Union[npt.NDArray[np.inexact], typing.Sequence[npt.NDArray[np.inexact]]` to highlight that the constants shouldn't be integers?

Reply:
I really don't know how NumPy arrays should be properly typed. I would prefer to not change this in this PR, and in a separate PR fix the typing across the boards.